docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,422 @@ overview page.
+consumes a lot of resources.
+
+Implementation Details
+----------------------
+
+Deployment
+~~~~~~~~~~
+
+All the components will be deployed using Kubernetes manifests through Salt
+inside a ``metalk8s-logging`` Namespace.
+
+As we already did for other components (e.g. Prometheus Operator), the
+manifests for Loki and Fluent Bit will be generated statically by rendering
+the `loki-stack helm chart`_::
+
+  helm repo add loki https://grafana.github.io/loki/charts
+  helm repo update
+  helm fetch -d charts --untar loki/loki-stack
+  ./charts/render.py loki-stack --namespace metalk8s-logging \
+    charts/loki-stack.yaml charts/loki-stack/ \
+    > salt/metalk8s/addons/loki-stack/deployed/charts.sls
+
+.. _loki-stack helm chart: https://github.com/grafana/loki/tree/master/production/helm
+
+We will use the same approach for Fluentd, using the `fluentd helm chart`_::
+
+  helm fetch -d charts --untar stable/fluentd
+  ./charts/render.py fluentd --namespace metalk8s-logging charts/fluentd.yaml \
+    charts/fluentd/ > salt/metalk8s/addons/fluentd/deployed/charts.sls
+
+.. _fluentd helm chart: https://hub.helm.sh/charts/stable/fluentd
+
+Container images must be added in the buildchain
+``buildchain/buildchain/image.py`` and then replaced in
+``charts/loki-stack.yaml`` and ``charts/fluentd.yaml`` using the
+``__image__()`` macro to be replaced by jinja syntax when rendering the
+charts and then dynamically computed at run time.
+
+Fluent Bit
+""""""""""
+
+Fluent Bit will be deployed as a DaemonSet, because we need one collector on
+each control or workload plane node to be able to collect logs from both the
+Kubernetes platform and the applications that run on top of it.
+
+Loki
+""""
+
+Loki will be deployed as a StatefulSet as it needs persistent volumes to write
+the logs. It will be running only on control plane nodes and we need at least
+2 replicas (except on a single node platform) of it to ensure its high
+availability.
+
+Loki HA
+_______
+
+Since we're using filesystem to store Loki's data, we have basically 2 ways
+for having multiple Loki instances running at the same time, with the same
+set of data.
+
+Either we do like Prometheus and we store everything on every instance of
+Loki, but it means we raise the storage, RAM and CPU needs for each additionnal
+instance, either we use a `new experimental feature`_ from Loki 1.5.0 where
+Loki ingesters use a hash ring and talk between them to route the queries to
+the right one. This approach needs an external KV store to work such as
+etcd or Consul.
+
+.. _new experimental feature: https://github.com/grafana/loki/blob/master/docs/operations/storage/filesystem.md#new-and-very-experimental-in-150-horizontal-scaling-of-the-filesystem-store
+
+.. note::
+
+  What do we do ? Do we use metalk8s etcd to store loki ring ? :/
+
+  Or do we add a new instance of etcd or consul ?
+
+  This feature is marked as experimental, should we even use it ?
```

Comment:
Looks very prone to changes if you ask me :/

Reply:
I agree, and probably bugged too.

Reply:
Since the Loki part of MetalK8s is only here for short term retention to me it's ok to store everything on every instance of Loki.

Reply:
As discussed offline, we want to consider our infra services as simple _poor man's solutions_, and document how to get metrics/logs/alerts/... out of the cluster into other systems (e.g. Splunk for logs).

Reply:
Ok, good to me, just wanted to be sure everyone is aware and agree on this.